lambdas/reconcile-migrated-vcs/src/main/java/uk/gov/di/ipv/core/reconcilemigratedvcs/ReconcileMigratedVcsHandler.java

```diff
@@ -0,0 +1,413 @@
+            processedIdentities.add(userId);
+            return;
+        }
+
+        // Fetch VCs from tactical
+        List<String> tacticalVcsStrings;
+        try {
+            tacticalVcsStrings =
+                    tacticalDataStore.getItems(userId).stream()
+                            .map(VcStoreItem::getCredential)
+                            .sorted()
+                            .toList();
+        } catch (Exception e) {
+            logError("Error reading from tactical", hashedUserId, e);
+            reconciliationReport.incrementFailedTacticalRead(hashedUserId);
+            reconciliationReport.incrementIdentitiesFullyProcessed();
+            processedIdentities.add(userId);
+            return;
+        }
+
+        // Compare VCs
+        if (!evcsVcsStrings.equals(tacticalVcsStrings)) {
+            logVcDifferences(evcsVcsStrings, tacticalVcsStrings, hashedUserId);
+            reconciliationReport.incrementDifferenceInVcs(hashedUserId);
+            reconciliationReport.incrementIdentitiesFullyProcessed();
+            processedIdentities.add(userId);
+            return;
+        } else {
+            reconciliationReport.incrementVcsMatch();
+        }
+
+        if (reconciliationReport.isCheckSignatures() || reconciliationReport.isCheckP2()) {
+            // Parse VCs from EVCS response
+            List<VerifiableCredential> evcsVcs = new ArrayList<>();
+            for (var vcString : evcsVcsStrings) {
+                try {
+                    var jwt = SignedJWT.parse(vcString);
+                    var cri = configService.getCriByIssuer(jwt.getJWTClaimsSet().getIssuer());
+                    evcsVcs.add(VerifiableCredential.fromValidJwt(userId, cri, jwt));
+                } catch (ParseException | CredentialParseException | NoCriForIssuerException e) {
+                    logError("Error parsing VC from EVCS", hashedUserId, e);
+                    reconciliationReport.incrementFailedToParseEvcsVcs(hashedUserId);
+                    reconciliationReport.incrementIdentitiesFullyProcessed();
+                    processedIdentities.add(userId);
+                    return;
+                }
+            }
+
+            if (reconciliationReport.isCheckSignatures()) {
+                // Validate VC signatures
+                checkSignatures(evcsVcs, hashedUserId);
+            }
+
+            if (reconciliationReport.isCheckP2()) {
+                // Check we've still got a P2 identity
+                checkIfP2(evcsVcs, hashedUserId);
+            }
+        }
+
+        processedIdentities.add(userId);
+        reconciliationReport.incrementIdentitiesFullyProcessed();
+
+        checkTimeout(context, hashedUserId);
+    }
+
+    private void checkSignatures(List<VerifiableCredential> evcsVcs, String hashedUserId) {
+        var identityFailedSigValidation = false;
+        for (var vc : evcsVcs) {
+            try {
+                if (!validSignature(vc, hashedUserId)) {
```

Comment:
Will the `validSignature` method deem `exp`ired VCs as still valid and not have a maxAge restriction

Reply:
Yeah, we're not inspecting anything other than the signature. We have other validators for inspecting the claims but we're not using them here.